rpc/item.c

```diff
@@ -540,6 +576,13 @@ M0_INTERNAL void m0_rpc_item_xid_assign(struct m0_rpc_item *item)
 		item->ri_header.osr_xid = rpc_item_needs_xid(item) ?
 					  ++item->ri_session->s_xid :
 					  UINT64_MAX;
+		if (rpc_item_needs_xid(item)) {
+			M0_LOG(M0_DEBUG, "xidl_tlist_add_tail(), item="ITEM_FMT,
+			       ITEM_ARG(item));
+			xidl_tlist_add_tail(&item->ri_session->s_xid_list,
+					    item);
+			item->ri_xid_assigned_here = true;
+		}
```

Comment:
Can you re-organize this like:
```
if if (rpc_item_needs_xid(item)) {
        item->ri_header.osr_xid = ++item->ri_session->s_xid;
        xidl_tlist_add_tail(&item->ri_session->s_xid_list,
                                     item);
        item->ri_xid_assigned_here = true;
} else {
        item->ri_header.osr_xid = UINT64_MAX;
}
```

Reply:
I had tried. The assignment became less readable because of `M0_LOG()` just before `xidl_tlist_add_tail()` - it distracts from how the assignment is done.